src/components/TransactionDialog/TransactionDialog.tsx

```diff
@@ -15,57 +34,139 @@ export type TransactionDialogProps = Pick<ActionDialogProps, 'className'> & {
     description:
       "Please wait till all your assets get processed. This can take up to 1 minute, depending on asset size and your machine's computing power.",
-    tooltip: '',
+    tooltip: 'Processing',
+    animation: {
+      data: firstStepAnimation,
+      size: {
+        width: 216,
+        height: 216,
+      },
+      loop: true,
+    },
   },
   [ExtrinsicStatus.Unsigned]: {
-    title: 'Waiting for signature...',
+    title: 'Sign in Polkadot',
     description: 'Please sign the transaction using the Polkadot browser extension.',
     tooltip: 'Signature',
+    animation: {
+      data: secondStepAnimation,
+      size: {
+        width: 288,
+        height: 216,
+      },
+      loop: false,
+    },
   },
   [ExtrinsicStatus.Signed]: {
-    title: 'Waiting for confirmation...',
+    title: 'Processing transaction',
     description:
       'Your transaction has been signed and sent. Please wait for the blockchain confirmation. This should take about 15 seconds.',
     tooltip: 'Confirmation',
+    animation: {
+      data: thirdStepAnimation,
+      size: {
+        width: 216,
+        height: 216,
+      },
+      loop: true,
+    },
   },
   [ExtrinsicStatus.Syncing]: {
-    title: 'Waiting for data propagation...',
+    title: 'Propagating changes',
     description:
       "Your transaction has been accepted and included into the blockchain. Please wait till it's picked up by the indexing node. This should take up to 15 seconds.",
     tooltip: 'Propagation',
+    animation: {
+      data: fourthStepAnimation,
+      size: {
+        width: 216,
+        height: 216,
+      },
+      loop: true,
+    },
   },
 }
 
 export const TransactionDialog: React.FC<TransactionDialogProps> = ({ status, onClose, ...actionDialogProps }) => {
+  const [polkadotLogoVisible, setPolkadotLogoVisible] = useState(false)
   const stepDetails =
-    status != null && status !== ExtrinsicStatus.Error && status !== ExtrinsicStatus.Completed
-      ? TRANSACTION_STEPS_DETAILS[status]
+    status != null && status !== ExtrinsicStatus.Error
+      ? TRANSACTION_STEPS_DETAILS[status === ExtrinsicStatus.Completed ? ExtrinsicStatus.Syncing : status]
       : null
 
-  const canCancel = status === ExtrinsicStatus.ProcessingAssets || status === ExtrinsicStatus.Unsigned
+  useEffect(() => {
+    if (status) {
+      setPolkadotLogoVisible(false)
+    }
+  }, [status])
+
+  const canCancel =
+    status === ExtrinsicStatus.ProcessingAssets ||
+    status === ExtrinsicStatus.Unsigned ||
+    status === ExtrinsicStatus.Completed
+
+  const transactionSteps = Object.values(TRANSACTION_STEPS_DETAILS)
 
-  const transactionStepsWithoutProcessingAssets = Object.values(TRANSACTION_STEPS_DETAILS).filter(
-    (step) => step.title !== TRANSACTION_STEPS_DETAILS[ExtrinsicStatus.ProcessingAssets].title
-  )
   return (
     <ActionDialog
       showDialog={!!stepDetails}
       secondaryButton={{ text: 'Cancel', onClick: onClose, disabled: !canCancel }}
```

Comment:
One thing which I don't like is that when the transaction is finished we are still showing the `Cancel` button which could be wrongly interpreted by a user. 
In the prototype, we automatically close the dialog with the disabled cancel button. We could do that, but it will require adding some delay(we still want to display completed animation). 

On the other hand, I think we could just show `Close` button: 
```suggestion
      secondaryButton={{
        text: status === ExtrinsicStatus.Completed ? 'Close' : 'Cancel',
        onClick: onClose,
        disabled: !canCancel,
      }}
```
And to be honest, I'm fine with the second option, but I'll consult this with a designer and let you know.

Reply:
Ok, talked to Adam, and we should automatically close the dialog after delay.